apps/costcontrol/test/unit/widget_startup_test.js

```diff
@@ -461,4 +453,82 @@ suite('Widget Startup Modes Test Suite >', function() {
     AirplaneModeHelper._status = 'disabled';
     Widget.init();
   });
+
+  test('DSDS normal start-up after sim switch from dataUseOnly to prepaid mode',
+    function(done) {
+      var fakePrePaidResult = {
+        datausage: requestDataUsageResult,
+        balance: resultRequestPrepaid
+      };
+
+      function assertPrepaidLayout() {
+        var mobileData =
+          Formatting.roundData(fakePrePaidResult.datausage.data.mobile.total);
+        var mobileDataText = _('magnitude', {
+          value: mobileData[0],
+          unit: mobileData[1]
+        });
+        var balanceDataText = _('currency', {
+          value: fakePrePaidResult.balance.data.balance,
+          currency: fakePrePaidResult.balance.data.currency
+        });
+
+        assertNonDataUseOnlyLayout(mobileDataText, balanceDataText);
+      }
+
+      this.sinon.stub(SimManager, 'isMultiSim').returns(true);
+      // SIM1 as DATA_USAGE_ONLY
+      var fakeResultDataUsage = {
+        datausage: requestDataUsageResult
+      };
+      var costControlConfig  = newCostControlRequest(fakeResultDataUsage);
+      setupCardState(costControlConfig);
+      var ftePending = false;
+      setupConfig('DATA_USAGE_ONLY', ftePending);
+
+      // Waiting for the first startup (DATA_USE_ONLY mode)
+      window.addEventListener('hashchange', function _onHashChange(evt) {
+        if (window.location.hash.split('#')[1] === 'updateDone') {
+          window.removeEventListener('hashchange', _onHashChange);
+          // right Panel is hidden when data_usage_only mode is loaded
+          assert.ok(rightPanel.hidden);
+
+          this.sinon.stub(window.CostControl, 'getInstance')
```

Comment:
The assert above is enough to check the layout. There is no need for checking or mocking up results.